### ISSUE TITLE
Error while trying to PutObject into Localstack S3

### ISSUE THREAD

**Issue by fernandosiq** [localstack/localstack#455]
Hi,

I've been running localstack in docker, and have been having success using it with Ruby AWS SDK.
Lately I've tried using it in Java AWS SDK and got an error:

```
2017-11-10T12:42:51:ERROR:localstack.services.generic_proxy: Error forwarding request: string index out of range Traceback (most recent call last):

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 162, in forward
   path=path, data=data, headers=forward_headers)

 File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 383, in forward_request
 modified_data = strip_chunk_signatures(data)

  File "/opt/code/localstack/localstack/services/s3/s3_listener.py", line 252, in strip_chunk_signatures
if data_new[0] in (10, 13):

IndexError: string index out of range
```

I went through the code and saw that it's trying to strip the chunck signatures, compared the requests that work (Ruby SDK):

**HEADERS:**
```
Content-Type:
Accept-Encoding:
User-Agent: aws-sdk-ruby3/3.6.0 ruby/2.2.8 x86_64-linux aws-sdk-s3/1.5.0
Expect: 100-continue
Content-Md5: 4HkQoGoIbIO6QYJ6oAsm7Q==
Host: 10.0.75.1:9000
X-Amz-Date: 20171110T130937Z
X-Amz-Content-Sha256: dc460da4ad72c482231e28e688e01f2778a88ce31a08826899d54ef7183998b5
Authorization: AWS4-HMAC-SHA256 Credential=abc/20171110/localhost/s3/aws4_request, SignedHeaders=content-md5;expect;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=acbf24709e49fdd2fd22965d928e1abd440bbdffb4c920a16cc140bac5e05ae0
Content-Length: 4
Accept: */*
```
**BODY:**
`asd`

Note that this asd is the content of the file.

with the one that doesn't (Java SDK) 1.11.180:

**HEADERS:**
```
x-amz-content-sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD
Authorization: AWS4-HMAC-SHA256 Credential=abc/20171110/localhost/s3/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-retry;content-length;content-md5;content-type;host;user-agent;x-amz-content-sha256;x-amz-date;x-amz-decoded-content-length, Signature=53456e284daeb26c9dc442e21659241848a22979a1f0da0d81d90e5a0658a222
X-Amz-Date: 20171110T124251Z
User-Agent: aws-sdk-java/1.11.180 Windows_10/10.0 Java_HotSpot(TM)_64-Bit_Server_VM/25.152-b16/1.8.0_152
amz-sdk-invocation-id: ec127ebc-81da-0d98-0cc7-a260c92a1ba5
x-amz-decoded-content-length: 0
amz-sdk-retry: 3/122/485
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
Content-Type: application/octet-stream
Content-Length: 86
Connection: Keep-Alive
Expect: 100-continue
```

**BODY:**
`0;chunk-signature=55de8c65df62554f8a104e5b23603684cae6e63dcbba8bac45c3b9ae99e3ad36`

**Comment by fernandosiq** [comment]
I was able to work around it by disabling the chunkedEncoding.

**Comment by whummer** [comment]
Thanks for reporting @fernandosiq . This should be fixed in #469 - can you please give it a try with the latest version? Thanks

**Comment by fernandosiq** [comment]
It's working now, thanks @whummer
